lib/balance-tx/lib/main/Cardano/Write/Eras.hs

```diff
@@ -0,0 +1,55 @@
+-- |
+-- Copyright: © 2024 Cardano Foundation
+-- License: Apache-2.0
+--
+module Cardano.Write.Eras
+    (
+    -- * Eras
+      BabbageEra
+    , ConwayEra
+
+    -- ** RecentEra
+    , RecentEra (..)
+    , IsRecentEra (..)
+    , CardanoApiEra
+    , toRecentEra
+    , fromRecentEra
+    , MaybeInRecentEra (..)
+    , LatestLedgerEra
+    , RecentEraConstraints
+    , allRecentEras
+
+    -- ** Existential wrapper
+    , AnyRecentEra (..)
+    , toAnyCardanoEra
+    , fromAnyCardanoEra
+
+    -- ** Helpers for cardano-api compatibility
+    , cardanoEra
+    , shelleyBasedEra
+    , ShelleyLedgerEra
+    , cardanoEraFromRecentEra
+    , shelleyBasedEraFromRecentEra
+    ) where
+
+import Internal.Cardano.Write.Eras
+    ( AnyRecentEra (..)
+    , BabbageEra
+    , CardanoApiEra
+    , ConwayEra
+    , IsRecentEra (..)
+    , LatestLedgerEra
+    , MaybeInRecentEra (..)
+    , RecentEra (..)
+    , RecentEraConstraints
+    , ShelleyLedgerEra
+    , allRecentEras
+    , cardanoEra
+    , cardanoEraFromRecentEra
+    , fromAnyCardanoEra
+    , fromRecentEra
+    , shelleyBasedEra
+    , shelleyBasedEraFromRecentEra
+    , toAnyCardanoEra
+    , toRecentEra
+    )
```

Comment:
I'm not too fond of duplicating this interface (friction to change, risk of sections getting out of sync etc), but we can go with it for now

Reply:
Yes. The idea is that the public interface, that is `Cardano.Write.Eras`, is the source of truth that we need to curate, and that `Internal.Cardano.Write.Eras` is an internal module that we can arrange as we please, including removal. For the time being, I'd like to invest into curation and avoid upsetting the internal arrangement of `balance-tx` too much.

I think the original idea behind the `:internal` sublibrary was that the public library can choose not to export certain internals, such as constructors for abstract data types, but that these internals are still available when unit testing the internal library.

Reply:
For the time being, I'd like to invest into curation and avoid upsetting the internal arrangement of balance-tx too much.

👍